server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/csv/DominionCVRExportParserTests.java

```diff
@@ -0,0 +1,585 @@
+/*
+Democracy Developers IRV extensions to colorado-rla.
+
+@copyright 2024 Colorado Department of State
+
+These IRV extensions are designed to connect to a running instance of the raire 
+service (https://github.com/DemocracyDevelopers/raire-service), in order to 
+generate assertions that can be audited using colorado-rla.
+
+The colorado-rla IRV extensions are free software: you can redistribute it and/or modify it under the terms
+of the GNU Affero General Public License as published by the Free Software Foundation, either
+version 3 of the License, or (at your option) any later version.
+
+The colorado-rla IRV extensions are distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.csv;
+
+import au.org.democracydevelopers.corla.model.ContestType;
+import au.org.democracydevelopers.corla.testUtils;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.testcontainers.containers.PostgreSQLContainer;
+import us.freeandfair.corla.csv.DominionCVRExportParser;
+import us.freeandfair.corla.model.*;
+import us.freeandfair.corla.persistence.Persistence;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.testng.annotations.*;
+
+import static au.org.democracydevelopers.corla.testUtils.*;
+import static org.testng.Assert.*;
+
+import static us.freeandfair.corla.query.CastVoteRecordQueries.getMatching;
+import static us.freeandfair.corla.query.ContestQueries.forCounties;
+import static us.freeandfair.corla.query.CountyQueries.fromString;
+
+/**
+ * Test a variety of IRV-related parsing issues for CVRs, including:
+ * - some basic small test cases that can be verified by eye,
+ * - a constructed test with invalid IRV ballots to check they are accepted and properly
+ *   interpreted.
+ * - a real example from Boulder '23, with a mix of IRV and plurality contests
+ * - some examples with invalid headers, to ensure they are rejected (though most of these are
+ *   tested in IRVHeadersParserTests.java and are not repeated here),
+ *   plus a bad plurality "Vote For= " with a non-integer.
+ * - an examples to test the broader class of Write In strings.
+ */
+public class DominionCVRExportParserTests {
+
+  /**
+   * Class-wide logger
+   */
+  private static final Logger LOGGER = LogManager.getLogger(DominionCVRExportParserTests.class);
+
+  /**
+   * Container for the mock-up database.
```

Comment:
Should "postgres:15-alpine" be hardcoded here? Can this be something placed in a properties file somewhere? Similarly, all the other names (eg "corla"), should they be hardcoded?

Reply:
Agree, they probably shouldn't be - I left it this way pending some discussion with @umbernhard and @charliecarlton about how we want to structure all the tests. So this is certainly expected to change, but I didn't bother cleaning it up yet.